packages/aragon-wrapper/src/templates/index.js

```diff
@@ -26,49 +27,118 @@ const templates = {
     params: [
       'name', // string
       'signers', // array of addresses
-      'neededSignatures', // number of signatures need, must be > 0 and <= signers.length
-    ],
-  },
+      'neededSignatures' // number of signatures need, must be > 0 and <= signers.length
+    ]
+  }
 }
 
 const Templates = (web3, apm, from) => {
   const minGasPrice = web3.utils.toWei('20', 'gwei')
   const newToken = async (template, name) => {
-    const receipt = await template.methods.newToken(name, name).send({ from, gas: 4e6, gasPrice: minGasPrice })
-    return receipt.events.DeployToken.returnValues
+    const progress = new Subject()
+    const events = template.methods.newToken(name, name).send({ from, gas: 4e6, gasPrice: minGasPrice })
+
+    events.on('transactionHash', (transactionHash) => {
+      progress.next({
+        transaction: 'TOKEN',
+        status: 'SIGNED',
+        meta: {
+          transactionHash
+        }
+      })
+    }).on('receipt', ({ transactionHash, events }) => {
+      const tokenAddress = events.DeployToken.returnValues.token
+      progress.next({
+        transaction: 'TOKEN',
+        status: 'MINED',
+        meta: {
+          transactionHash,
+          address: tokenAddress
+        }
+      })
+      progress.complete()
+    }).on('error', (error, { transactionHash }) => {
+      progress.next({
+        transaction: 'TOKEN',
+        status: 'ERROR',
+        meta: {
+          transactionHash,
+          message: error.message
+        }
+      })
+      progress.complete()
+    })
+
+    return progress
   }
 
   const newInstance = async (template, name, params) => {
-    const receipt = await template.methods.newInstance(name, ...params).send({ from, gas: 6.6e6, gasPrice: minGasPrice })
-    return receipt.events.DeployInstance.returnValues
+    const progress = new Subject()
+    const events = template.methods.newInstance(name, ...params).send({ from, gas: 6.9e6, gasPrice: minGasPrice })
```

Comment:
@izqui Don't remember the exact setting we wanted for gas limit, but I remember metamask was flaking at 6.9.

Reply:
When I resolve the conflict it's going to use gas estimation